kernel/threads.c

```diff
@@ -143,3 +146,42 @@ void block_current_thread (void) {
   running_tcb->state = TS_BLOCKED;
   schedule();
 }
+
+int fork_ret = 0;
+
+int clone_thread (uint64_t fork_rsp) {
+  /* This tells us how much of the kernel stack to copy to the new thread */
+  uint64_t stack_depth = (uint64_t) running_tcb->stack_top - fork_rsp;
+  for (int i = 0; i < NUMTHREADS; i++) {
+    if (tcbs[i].state == TS_NONEXIST) {
```

Comment:
George mentioned this as a nice-to-have to me IRL as well, so I got around to doing it.